cannot fetch/process workflows/jobs which failed with github's startup_failure

**Description + reprodcution**

I'm setting up the docker image, pointing it at one of my (private) repos.
I had some trouble with setting up the workflows in said repo, so many of the runs I triggered did not start because of workflow parsing errors.
During startup of the container, I see the following logs:
```
# a bunch of
2023-04-17T11:39:02.922Z  INFO 1 --- [nPool-worker-11] d.o.p.g.a.s.job.GithubJobDetailsService  : Fetching run details for Workflow(name=check_for_release, repoName=test-work
flow-viewer, id=54272134)
...
2023-04-17T11:39:03.459Z ERROR 1 --- [nPool-worker-10] d.o.p.g.a.s.job.GithubJobDetailsService  : org.springframework.web.client.RestClientException: Error while extracting response for type [class de.otto.platform.gitactionboard.adapters.service.job.WorkflowsRunDetailsResponse] and content type [application/json;charset=utf-8]
...
java.util.concurrent.CompletionException: org.springframework.web.client.RestClientException: Error while extracting response for type [class de.otto.platform.gitactionboard.adapters.service.job.WorkflowsRunDetailsResponse] and content type [application/json;charset=utf-8]
Caused by: com.fasterxml.jackson.databind.exc.InvalidFormatException: Cannot deserialize value of type `de.otto.platform.gitactionboard.adapters.service.job.RunConclusion` from String "startup_failure": not one of the values accepted for Enum class: [skipped, cancelled, success, timed_out, neutral, action_required, failure]
 at [Source: (org.springframework.util.StreamUtils$NonClosingInputStream); line: 1, column: 341] (through reference chain: de.otto.platform.gitactionboard.adapters.service.job.WorkflowsRunDetailsResponse["workflow_runs"]->java.util.ArrayList[0]->de.otto.platform.gitactionboard.adapters.service.job.WorkflowsRunDetailsResponse$WorkflowRunDetails["conclusion"])
 ```

**Additional context**
Tested with docker image `3.1.0` and `latest`.

--- a/backend/src/main/java/de/otto/platform/gitactionboard/adapters/service/job/RunConclusion.java
+++ b/backend/src/main/java/de/otto/platform/gitactionboard/adapters/service/job/RunConclusion.java
@@ -11,6 +11,9 @@ public enum RunConclusion {
   @JsonProperty("failure")
   FAILURE,
 
+  @JsonProperty("startup_failure")
+  STARTUP_FAILURE,
+
   @JsonProperty("neutral")
   NEUTRAL,
 
@@ -32,7 +35,7 @@ public enum RunConclusion {
     }
     return switch (conclusion) {
       case SUCCESS, SKIPPED -> Status.SUCCESS;
-      case FAILURE, CANCELLED, TIMED_OUT -> Status.FAILURE;
+      case FAILURE, STARTUP_FAILURE, CANCELLED, TIMED_OUT -> Status.FAILURE;
       case ACTION_REQUIRED -> Status.EXCEPTION;
       default -> Status.UNKNOWN;
     };

--- a/backend/src/test/java/de/otto/platform/gitactionboard/adapters/service/job/RunConclusionTest.java
+++ b/backend/src/test/java/de/otto/platform/gitactionboard/adapters/service/job/RunConclusionTest.java
@@ -23,7 +23,7 @@ class RunConclusionTest {
   @ParameterizedTest(name = "should give FAILURE status for {0} conclusion")
   @EnumSource(
       value = RunConclusion.class,
-      names = {"TIMED_OUT", "CANCELLED", "FAILURE"})
+      names = {"TIMED_OUT", "CANCELLED", "FAILURE", "STARTUP_FAILURE"})
   void shouldGiveFailureStatus(RunConclusion conclusion) {
     assertThat(RunConclusion.getStatus(conclusion)).isEqualTo(Status.FAILURE);
   }

